dot/core/service.go

```diff
@@ -550,3 +554,61 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	rt.SetContextStorage(ts)
 	return rt.Metadata()
 }
+
+// QueryStorage returns the key-value data by block based on `keys` params
+// on every block starting `from` until `to` block, if `to` is not nil
+func (s *Service) QueryStorage(from *common.Hash, to *common.Hash, keys ...string) (map[common.Hash]QueryKeyValueChanges, error) {
+	if from == nil {
+		return nil, errors.New("cannot query data without a starting block hash")
+	}
```

Comment:
why is `from` a pointer, if it can never be nil?

Reply:
Good question, I can update the `from` to be a value of `common.Hash` instead of a pointer, could `to` stay as a pointer?

Reply:
yes, that would make sense!